src/Provider/ES.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Checkdomain\Holiday\Provider;
+
+/**
+ * Spanish holiday provider
+ *
+ * @author Remi Schillinger <remi.schillinger@gmail.com>
+ * @since 2020-12-22
+ */
+class ES extends AbstractEaster
+{
+    const STATE_AN = 'Andalucía';
+    const STATE_AR = 'Aragón';
+    const STATE_CA = 'Cataluña';
+    const STATE_CE = 'Ceuta';
+    const STATE_CL = 'Castilla y León';
+    const STATE_CM = 'Castilla La Mancha';
+    const STATE_CN = 'Cantabria';
+    const STATE_CV = 'Comunidad Valenciana';
+    const STATE_EX = 'Extremadura';
+    const STATE_GA = 'Galicia';
+    const STATE_IB = 'Islas Baleares';
+    const STATE_IC = 'Islas Canarias';
+    const STATE_LR = 'La Rioja';
+    const STATE_MA = 'Madrid';
+    const STATE_ME = 'Melilla';
+    const STATE_MU = 'Murcia';
+    const STATE_PA = 'Principado de Asturias';
+    const STATE_PV = 'País Vasco';
+
+    /**
+     * @param int $year
+     *
+     * @return mixed
+     */
+    public function getHolidaysByYear($year)
+    {
+        $easter = $this->getEasterDates($year);
+
+        $holidays = array(
+            '01-01' => $this->createData('Año Nuevo'),
+            '01-06' => $this->createData('Epifanía'),
+            '02-28' => $this->createData('Día de la comunidad', array(
+                self::STATE_AN,
+            )),
+            '03-01' => $this->createData('Día de la comunidad', array(
+                self::STATE_IB,
+            )),
+            '03-19' => $this->createData('San José', array(
+                self::STATE_CM,
+                self::STATE_CV,
+                self::STATE_GA,
+                self::STATE_IB,
+                self::STATE_MA,
+                self::STATE_ME,
+                self::STATE_MU,
+                self::STATE_PA,
+            )),
```

Comment:
Local/regional holidays change from one year to another. It seems impossible to integrate them automatically since they are decided arbitrarily  

> Public holidays celebrated in Spain include a mix of religious (Roman Catholic), national and regional observances. Each municipality is allowed to have a maximum of 14 public holidays per year; a maximum of nine of these are chosen by the national government and at least two are chosen locally.

> If one of the "national holidays" happens to fall on a Sunday the regional governments — the autonomous communities of Spain — can choose an alternate holiday or they can allow local authorities to choose. In practice, except for holidays falling on a Sunday, the regional governments can choose up to three holidays per year; or they can choose fewer to allow for more options at the local level.

You can see it for instance at : https://www.calendarioslaborales.com/